CRM/MembershipExtras/Hook/PostProcess/UpdateSubscription.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+use CRM_MembershipExtras_Service_InstallmentReceiveDateCalculator as InstallmentReceiveDateCalculator;
+
+/**
+ * Post-processes Recurring Conribution Update form.
+ */
+class CRM_MembershipExtras_Hook_PostProcess_UpdateSubscription {
+
+  /**
+   * Form to be processed.
+   *
+   * @var \CRM_Contribute_Form_UpdateSubscription
+   */
+  private $form;
+
+  public function __construct(CRM_Contribute_Form_UpdateSubscription $form) {
+    $this->form = $form;
+    $this->recurringContributionID = CRM_Utils_Request::retrieve('crid', 'Integer', $this->form, TRUE);

```

Comment:
same here, the following two lines can be moved to new method called $this->setRecurringContribution().

Also no need to store both recurringContributionID  and recurringContribution properties, the latter is enough and you can access the ID using $this->recurringContribution['id']